### DISCUSSION TITLE
IFUS y211ga home:  hack version 0.2.5 and fw base 12 (problem)

### DISCUSSION THREAD

**Discussion by Giuserver** [roleoroleo/yi-hack-Allwinner-v2#547] [Q&A]
Hello!
My camera IFUSY65****** HOME 1080p (model y21ga) rebooting it self continuously after upgrade to version 0.2.5. and FW Base 12.0.19***

@roleoroleo any idea's or it's possible back to 9.0.19* ?

Thanks,
Giuseppe

**Comment by roleoroleo** [comment]
About reboot check the faq page in the wiki. Maybe you can fix.
Or you can downgrade to the previous fw following the unbrick guide.

**Reply by Giuserver** [reply]
if I format the SD and reinsert into camera, work fine. Also, i can enter into camera with telnet port (23), maybe can help?

**Comment by roleoroleo** [comment]
At the moment y21ga with 12.x is not supported.
I don't know how to fix the problem.

**Comment by roleoroleo** [comment]
Try to remove the file /tmp/sd/yi-hack/bin/cloudAPI_real
And overwrite /tmp/sd/yi-hack/bin/mqttv4 with this one:
[mqttv4.gz](https://github.com/roleoroleo/yi-hack-Allwinner-v2/files/10239911/mqttv4.gz)

**Reply by Giuserver** [reply]
> Try to remove the file /tmp/sd/yi-hack/bin/cloudAPI_real And overwrite /tmp/sd/yi-hack/bin/mqttv4 with this one: [mqttv4.gz](https://github.com/roleoroleo/yi-hack-Allwinner-v2/files/10239911/mqttv4.gz)

First of all, i got the wrong version. This camera is y211 (always IFUS) really sorry.
Anyway, i tried:
1) formatted the SD
2) extracted and copied in the SD the version 0.2.5 y211ga. Still work for a few seconds, after that starting rebooting
4) deleted the file /tmp/sd/yi-hack/bin/cloudAPI_real and overwrite the mqttv4 file whit yours in /tmp/sd/yi-hack/bin/mqttv4
The camera continue with rebooting each 10 seconds approximately.

Good for you @lohacker84 

Really thanks for your help @roleoroleo and sorry again. 
the actually (really bad) base version is: 12.0.37.0_202211241030

**Comment by lohacker84** [comment]
I used your files that I will never stop thanking you for this and I went back to 9.0.19.02_202007141512 because it couldn't be seen well at night.  it all worked perfectly!  Thank you so much for all the work you do for us!  you are great!

**Comment by roleoroleo** [comment]
I will release a new beta version to support new 12 firmwares.

**Reply by Giuserver** [reply]
So, 
I found the base version 11.0.36.7_202202170022 on the kami site.
I downloaded the original firmware from https://s3.us-west-1.wasabisys.com/firmware-release/y211ga/11.0.36.7_202202170022/one_y211ga.bin
from "original" link https://kamihome.com/firmware-release/y211ga/11.0.36.7_202202170022/one_y211ga.bin

formatted SD.
copy one_y211ga.bin into SD
inserted SD into camera
wait 3 minutes...Firmware 11 recoverd.
formatted SD
installed 0.25 hack..Perfect.
So, now I have base version: 11.0.36.7_202202170022
hack: 0.25

(of course no motion detection for Yi/Kami app but I don't care anymore from that apps..) anyway...i recover all services (thank to your hack) BUT:

the motion detection doesn't work correctly. This output from motion detection (I get it repeatedly new connection from camera)

1672247284: New connection from 192.168.1.6:52100 on port 1883.
1672247284: New client connected from 192.168.1.6:52100 as camera2 (p2, c1, k120).
1672247285: New connection from 192.168.1.6:52102 on port 1883.
1672247285: New connection from 192.168.1.6:52104 on port 1883.
1672247285: New client connected from 192.168.1.6:52102 as camera2_c (p2, c1, k15).
1672247285: Client camera2 already connected, closing old connection.
1672247285: New client connected from 192.168.1.6:52104 as camera2 (p2, c1, k120).
1672247285: Client camera2_c closed its connection.
1672247286: New connection from 192.168.1.6:52110 on port 1883.
1672247286: Client camera2 already connected, closing old connection.
1672247286: New client connected from 192.168.1.6:52110 as camera2 (p2, c1, k120).

Of course, the best would be recover 9.0.36 version but I can't find file .bin on Internet

Also, I noticed that .mp4 files recording like 
/record/2022Y12M28D17H/E114M18S42.mp4
The time zone seems wrong and the files are not 60 seconds long but always minus (between 45 and 60 seconds).
(I am in Rome, so i set +1 just in App, no time zone set in hack page) If I change the time zone in App (for example universal time zone) the file change name like: /record/2022Y12M28D17H/E015M18S47.mp4 But the problem I described before still remain

Giuseppe
and THANKS

**Reply by roleoroleo** [reply]
Motion detection problem is a know issue of the fw 11.x
I think I will not support this version because it's a long task and version 12 is already released.
If you want to try a new beta for fw 12 please let me know.
Motion detection should be fixed, about time zone I need to check.

**Reply by Giuserver** [reply]
@roleoroleo yes, sendme beta version, I'll try just right now

**Comment by roleoroleo** [comment]
This is the official update to fw 12.
[12.0.37.0_202211241030home_y211gam.zip](https://github.com/roleoroleo/yi-hack-Allwinner-v2/files/10316457/12.0.37.0_202211241030home_y211gam.zip)

This is the last beta.
[y211ga_0.2.5.tar.gz](https://github.com/roleoroleo/yi-hack-Allwinner-v2/files/10316454/y211ga_0.2.5.tar.gz)

**Reply by randyesperben** [reply]
Hi!

I've the same problem, reboot every 10 seconds avg.
I tried the last beta but the same issue.

Thanks for all your efforts.

**Reply by Giuserver** [reply]
Since I installed your firmware base and beta hack, when i insert an sd into camera (also empty) the camera start rebooting! (I tried also with different sd) Now the camera working just without sd!
I can enter in camera with telnet port 23 (of course, just without sd inserted)

**Reply by roleoroleo** [reply]
Please, try again with this new file:
[y211ga_0.2.5.tar.gz](https://github.com/roleoroleo/yi-hack-Allwinner-v2/files/10327157/y211ga_0.2.5.tar.gz)

**Reply by Giuserver** [reply]
So, i restart all procedures.
reset camera
Formatted sd
copy inside camera the new firmware.
Reboot and boom, everything come back normal.
Seems all services restored normally: rtsp, snapshot, motion detections ecc. 

**Bro you're on fire** 🔥

I will update you. I will do many tests for you. 

Thank you

**Reply by MaryCam1** [reply]
> This is the official update to fw 12. [12.0.37.0_202211241030home_y211gam.zip](https://github.com/roleoroleo/yi-hack-Allwinner-v2/files/10316457/12.0.37.0_202211241030home_y211gam.zip)
> 
> This is the last beta. [y211ga_0.2.5.tar.gz](https://github.com/roleoroleo/yi-hack-Allwinner-v2/files/10316454/y211ga_0.2.5.tar.gz)

works also for y21ga ?

**Reply by Giuserver** [reply]
> > This is the official update to fw 12. [12.0.37.0_202211241030home_y211gam.zip](https://github.com/roleoroleo/yi-hack-Allwinner-v2/files/10316457/12.0.37.0_202211241030home_y211gam.zip)
> > This is the last beta. [y211ga_0.2.5.tar.gz](https://github.com/roleoroleo/yi-hack-Allwinner-v2/files/10316454/y211ga_0.2.5.tar.gz)
> 
> works also for y21ga ?

NO. but you come back to 9.0, just follow this comment  https://github.com/roleoroleo/yi-hack-Allwinner-v2/discussions/547#discussioncomment-4412874
After that, download the original hack 0.2.5 from https://github.com/roleoroleo/yi-hack-Allwinner-v2/releases

**Reply by MaryCam1** [reply]
> > > This is the official update to fw 12. [12.0.37.0_202211241030home_y211gam.zip](https://github.com/roleoroleo/yi-hack-Allwinner-v2/files/10316457/12.0.37.0_202211241030home_y211gam.zip)
> > > This is the last beta. [y211ga_0.2.5.tar.gz](https://github.com/roleoroleo/yi-hack-Allwinner-v2/files/10316454/y211ga_0.2.5.tar.gz)
> > 
> > 
> > works also for y21ga ?
> 
> NO. but you come back to 9.0, just follow this comment [#547 (comment)](https://github.com/roleoroleo/yi-hack-Allwinner-v2/discussions/547#discussioncomment-4412874) After that, download the original hack 0.2.5 from https://github.com/roleoroleo/yi-hack-Allwinner-v2/releases

Thnx to Giuserver

@roleoroleo; will there also be an update to version 12 / y21ga in the future without having to downgrade to 9?

**Comment by evider2002** [comment]
@roleoroleo I tested your last version of the hack on fw 12, and indeed you miraculously fixed both the boot loop and no motion issues!!! 
There is though one more tweak necessary to make it perfect. 
In the oem fw of 9, 11 and 12, you need to choose either motion or AI, you cant have both, its either or.  
For example, in 9 when you turn on person detection, you are turning off any other motion, so if there is a non person motion, it won't come through. The same is with 12, when you turn on AI (person, vehicle, animal) you are turning off all the remaining non AI motion. This is a problem, because I want the benefit of AI but not at the expense of losing all the remaining non AI motion. 
This was not an issue with the hack on 9, somehow the way you had it set up, I was getting person AI alerts plus all the remaining non person alerts were coming through as "motion tracking triggered" alerts. It was the best of both worlds, I was able to have the cake and eat it. 
Please reimplement this into the 12 hack. Perhaps you can add an option to Motion Detection (0.0.0.0:8080/?page=camera_settings) something like "Motion + AI". Or maybe you can put back the "Motion Tracking sensor" option, so that I can still get motion alerts that way.

Thank you!

**Reply by evider2002** [reply]
I spend some more time testing, and I think I can explain how I was getting simultaneously AI and motion alerts with the hack on fw 9. 

The alert name for regular motion is called "motion detected". 
There is another similar motion called "motion tracking triggered", this is the motion alert intended for the YI Dome Camera, which can move and follow the motion. 
When AI is turned on (either "person" in 9, or "person, animal, vehicle" in 12) then it only shuts off "motion detected" alerts but it does not shut off "motion tracking triggered" alerts.
The unit I have is not the Yi Dome, what I have is the Yi home 1080p ai+.
However, in the public released hack for 9, you kept available the "motion tracking triggered" option. 
So even though its not an official alert available for my unit, but since you included it in the hack, I selected it anyway. 
By selecting it, I am able to get motion alerts (motion tracking triggered) even when AI is turned on. 

The problem is, that on the beta version for 12, you took out the "motion tracking triggered" option. 
Please put that option back in. 
Or maybe even better, you can figure out how to get "motion detected" alerts to work together with AI, just like "motion tracking triggered" works with AI. 

As it stands now, I prefer 9 - motion with only person AI, rather than 12 - no motion with all 3 AI. 
I am not willing to sacrifice motion for animal and vehicle AI.

Thank you. 

Here is a screenshot of both motion alerts in fw 9 (when AI is off).
![Screenshot_2023-01-03-15-38-51cropped](https://user-images.githubusercontent.com/118506981/210441955-a7ffffbf-bb4a-4ea0-9fa5-6a36d6cd4402.png)


And here is a screen shot of simultaneous AI and motion alerts in fw 9. 

![Screenshot_2023-01-03-16-15-02 cropped](https://user-images.githubusercontent.com/118506981/210442904-fafce4be-1b98-4b7b-bc02-f9127a861983.png)

**Reply by roleoroleo** [reply]
Motion tracking is not removed, it's simpy hidden.
If Yi didn't remove it, you can try to enable it manually, editing the file /tmp/sd/yi-hack/etc/camera.conf
`MOTION_TRACKING=yes`
and running the command
`ipc_cmd -o ON`
But if it doesn't work, it means that Yi removed the feature and you need to come back to fw 9.

**Comment by Giuserver** [comment]
@roleoroleo 
So i do many tests, like promise:
Maybe my problems coming from time zone but the camera try to connect continuously with mosquitto. 
So the motion detection doesn't work properly (I tried with private mode too (disabling cloud) same problem)


<img width="1082" alt="Screenshot 2023-01-03 alle 18 39 52" src="https://user-images.githubusercontent.com/97843369/210412403-ec0a87c5-efc6-4f2e-ba42-0fe924159464.png">
<img width="914" alt="Screenshot 2023-01-03 alle 18 40 31" src="https://user-images.githubusercontent.com/97843369/210412414-e20672a2-7d35-44d5-be7d-1d6035a0f141.png">

**Reply by roleoroleo** [reply]
When there is a disconnection problem with mosquitto, probably you are use 2 client with the same client id.
Check if this is your case: 2 cams or 2 processes with the same client id.

**Reply by Giuserver** [reply]
> When there is a disconnection problem with mosquitto, probably you are use 2 client with the same client id.
> Check if this is your case: 2 cams or 2 processes with the same client id.

I always use 1 camera at time for testing. Anyway, I changed default name but same problem, like you see the camera trying to connect with two different name at broker .

1672845680: mosquitto version 2.0.15 running
1672845735: New connection from 192.168.124.119:43736 on port 1883.
1672845735: New connection from 192.168.124.119:43738 on port 1883.
1672845735: New client connected from 192.168.124.119:43736 as camera1 (p2, c1, k120).
1672845735: New client connected from 192.168.124.119:43738 as camera1_c (p2, c1, k15).
1672845735: Client camera1_c closed its connection.
1672845737: New connection from 192.168.124.119:43744 on port 1883.
1672845737: New client connected from 192.168.124.119:43744 as camera1_c (p2, c1, k15).
1672845737: Client camera1_c closed its connection.
1672845737: New connection from 192.168.124.119:43746 on port 1883.

What about for videos recording so bad?
If I can do something for help i do, just tell me which classes use them. 
I have another camera IFUS y211ga with fw base 9.0 and 0.2.5 hack, maybe can help (by the way this camera also record video like E100M01S60.mp4 but correctly 1 minute long).
If for you is more easy and you have 9.0 fw base, just send me. It will be great too. 

Thank you.
Giuseppe

**Comment by roleoroleo** [comment]
Please try this new beta:
[y211ga_0.2.5.tar.gz](https://github.com/roleoroleo/yi-hack-Allwinner-v2/files/10360459/y211ga_0.2.5.tar.gz)

I changed many many lines of code.
This fw 12 is not so easy to understand...

**Reply by Giuserver** [reply]
Ok. So: first of all the web page in: http://ipcam:8080/?page=camera_settings. When i press the button to ON in Motion detection, save and refresh the page, the motion detection button remain OFF (I tried also reboot camera, but same); anyway i change MOTION_DETECTION=yes
in /tmp/sd/yi-hack/etc/camera.conf and MQTT=yes in /tmp/sd/yi-hack/etc/system.conf. After that, same problem on connections. I tried with and without "cloud mode" nothing change. I receive the notifications on yi app like in my mosquitto_sub when a motion it's detected (but badly). 
mosquitto broker still receive a lot of connection like before:
[...] 1673114284: New client connected from 192.168.43.140:40012 as camera1 (p2, c1, k120).
1673114284: New connection from 192.168.43.140:40010 on port 1883.
1673114284: Client camera1_c already connected, closing old connection. [...]

The record work bad too (like "pixeling") creating more file for every minute (also with "Save video when a motion is detected" it's on) 
It's like a two different users starting the same process for motion and record.

Thanks for all your effort, i hope this help a lot of users.
Giuseppe

**Reply by roleoroleo** [reply]
> Ok. So: first of all the web page in: http://ipcam:8080/?page=camera_settings. When i press the button to ON in Motion detection, save and refresh the page, the motion detection button remain OFF (I tried also reboot camera, but same);

With the fw 12 "Motion detection" is intended as generic motion detection (no AI), so if you enable it, all AI detections will be disabled.
And vice versa.

> Thanks for all your effort, i hope this help a lot of users. Giuseppe

Maybe there is a problem starting services, please run the ps command and post the output.

Please, try a reset to factory and reconfigure the cam from scratch.
I'm running this beta fw in my y21ga and it's working.

**Reply by Giuserver** [reply]
Ok. Yes, i do it (again). Maybe I explained myself wrong. I show you: 
<img width="1060" alt="Screenshot 2023-01-08 alle 15 14 42" src="https://user-images.githubusercontent.com/97843369/211201220-d2ffa380-a851-4778-8b8e-e3cce25df15c.png">
<img width="1135" alt="Screenshot 2023-01-08 alle 15 14 49" src="https://user-images.githubusercontent.com/97843369/211201223-fa37078e-db2a-464d-adda-0e15bbcc9577.png">


This happen when i refresh page or reboot the camera!

By the way, the most important thing it's I have y211ga not y21ga like i said here :https://github.com/roleoroleo/yi-hack-Allwinner-v2/discussions/547#discussioncomment-4498927 

Now i change also the title of this post.

[ps_output.log](https://github.com/roleoroleo/yi-hack-Allwinner-v2/files/10368636/ps_output.log)


Thank you

**Reply by roleoroleo** [reply]
Did you try to clear the cache of the browser? js code is changed.

I know your cam is a y211ga but the difference between y21ga and y211ga is very small.

**Reply by Giuserver** [reply]
Yes. I tried also change browser. In fact it seemed strange to me, because with the previous version (https://github.com/roleoroleo/yi-hack-Allwinner-v2/discussions/547#discussioncomment-4559771) i see that: 
<img width="1022" alt="Screenshot 2023-01-09 alle 11 15 59" src="https://user-images.githubusercontent.com/97843369/211285519-ffecac96-21fb-4e9e-8a6e-2a80c9e63d6a.png">

I tried also reinstalling the previous beta, formatted, reset camera and again install the last one beta, but in the last I have that page (https://github.com/roleoroleo/yi-hack-Allwinner-v2/discussions/547#discussioncomment-4625611).
By the way, if you've checked the ps_output.log you see the double service started (mqttv4)..

Thank you 😊

**Reply by roleoroleo** [reply]
Ok, I will check the code again.

About mqtt there is a bigger problem: all your processes are started twice:
```
 2062 root      1228 S    ./cloud
 2063 root      3564 S    ./p2p_tnp
 2064 root      3536 S    ./oss
 2066 root       744 S    ./watch_process
...
 2109 root      1056 S    mqttv4
 2114 root       776 S    mqtt-config
 2206 root      1228 S    ./cloud
 2207 root      3564 S    ./p2p_tnp
 2208 root      3536 S    ./oss
 2210 root       744 S    ./watch_process
 2258 root      1140 S    mqttv4
 2263 root       776 S    mqtt-config
```

Something is wrong in your config or in your system.sh

**Reply by Giuserver** [reply]
I understand. so, If you need something like backup, output command etc. Write here. I have another one camera y211ga with fw base 11 and hack 0.2.5. but for now i avoid to update that camera too for these reasons.

Thanks

**Reply by roleoroleo** [reply]
I'm close to completing a stable release.
I will send a new release to test asap.

**Reply by Giuserver** [reply]
When you ready. No rush.
Thanks

**Reply by blfriedman** [reply]
I also have an IFUS v11.0.36.7_202202170022 1080p cam.  It came with the v11 software (refurb), so I don't have a v9 to downgrade to, and it shows an upgrade available to v12.0.37.0_202211241030 which I have not updated to (yet).

I am willing to be an additional tester for whichever version (v11 or v12) you plan to release for hack 0.2.5.

My platform is to disable the yi cloud, disable motion detection and recording, and just to use ssh, ntp, http, and rtsp (no ftp, no telnet, no mqtt, no onvif).

**Reply by roleoroleo** [reply]
I will not support version 11; Only 9 or 12.
You can upgrade to version 12 if you want.
